### PR TITLE
[SPARK-23276][SQL][TEST] Enable UDT tests in (Hive)OrcHadoopFsRelationSuite

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
## What changes were proposed in this pull request?

Like Parquet, ORC test suites should enable UDT tests.

## How was this patch tested?

Pass the Jenkins with newly enabled test cases.

### CODE REVIEW

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcHadoopFsRelationSuite.scala [hunk 1]
```diff
@@ -34,11 +34,10 @@ class OrcHadoopFsRelationSuite extends HadoopFsRelationTest {
   override val dataSourceName: String =
     classOf[org.apache.spark.sql.execution.datasources.orc.OrcFileFormat].getCanonicalName
 
-  // ORC does not play well with NullType and UDT.
+  // ORC does not play well with NullType.
   override protected def supportsDataType(dataType: DataType): Boolean = dataType match {
     case _: NullType => false
     case _: CalendarIntervalType => false
-    case _: UserDefinedType[_] => false
```
Comment (dongjoon-hyun) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcHadoopFsRelationSuite.scala]: Both `OrcHadoopFsRelationSuite` and `HiveOrcHadoopFsRelationSuite` enables UDT tests.

**Review by HyukjinKwon — APPROVED** [review]
LGTM

**Comment by dongjoon-hyun** [comment]
Thank you for review, @HyukjinKwon .

**Comment by gatorsmile** [comment]
Thanks! Merged to master/2.3

**Comment by dongjoon-hyun** [comment]
Thank you, @HyukjinKwon  and @gatorsmile !

**Pull request closed without merging**